src/NHibernate/Hql/Ast/ANTLR/Tree/AggregateNode.cs

```diff
@@ -19,6 +20,19 @@ public AggregateNode(IToken token)
 		{
 		}
 
+		public string FunctionName
```

Comment:
I think this could be an overgeneralization and should be refactored into a node-specific code unless you see that it can be extended for functions other than `count` and `count_big`

Reply:
This may be extended for string concatenation, which is supported by some databases ([Sql server](https://docs.microsoft.com/en-us/sql/t-sql/functions/string-agg-transact-sql?view=sql-server-ver15), [MySql](https://dev.mysql.com/doc/refman/8.0/en/group-by-functions.html#function_group-concat), [Postgresql](https://www.postgresql.org/docs/9.1/functions-aggregate.html) and [Oracle](https://docs.oracle.com/database/121/SQLRF/functions101.htm#SQLRF30030)), where different function names are used (`LISTAGG`, `GROUP_CONCAT`, `STRING_AGG`).

Reply:
@maca88 can you provide some sort of prototype of string_agg? I'm dying to use this with linq provider

Reply:
To my knowledge it is not possible to implement it without modifying NHibernate source code. It is on my todo list for the next release, so unfortunately you will have to wait.